__tests__/app.test.js

```diff
@@ -83,6 +83,57 @@ describe("app.js GET requests", () => {
         })
     })
 
+    describe("GET /api/articles/:article_id", () => {
+
+        //Create random number within range of test data values for parametric endpoint testing
+        let article_id = Math.floor(Math.random()*(15-1))+1;
+
+    //Functionality tests
+        test("Status: 200 should return a single object", () => {
+            return supertest(app)
+            .get(`/api/articles/${article_id}`)
+            .then((result) => {
+                expect(200);
+                expect(result.body).toBeInstanceOf(Object);
+                expect(Array.isArray(result.body)).toBe(false);
+            })
+        })
+        test("returned article object has expected properties in expected formats", () => {
+            return supertest(app)
+            .get(`/api/articles/${article_id}`)
+            .then((result) => {
+                console.log(result.body);
+                expect(200);
+                expect(result.body).toHaveProperty("author", expect.any(String));
+                expect(result.body).toHaveProperty("title", expect.any(String));
+                expect(result.body).toHaveProperty("article_id", expect.any(Number));
+                expect(result.body).toHaveProperty("body", expect.any(String));
+                expect(result.body).toHaveProperty("topic", expect.any(String));
+                expect(result.body).toHaveProperty("created_at", expect.any(String));
+                expect(result.body).toHaveProperty("votes", expect.any(Number));
+                expect(result.body).toHaveProperty("article_img_url", expect.any(String));
+            })
+        })
+    //Error handling tests
+        test("Status: 400 and appropriate message if invalid article_id", () => {
+            return supertest(app)
+            .get(`/api/articles/banana`)
+            .then((result) => {
+                expect(400);
+                expect(result.body.msg).toBe("Invalid article id.")
+            })
+        })
+        test("Status 404 and appropriate message if valid but non-existent article_id", () => {
+            return supertest(app)
+            .get(`/api/articles/333`)
+            .then((result) => {
+                expect(404);
+                expect(result.body.msg).toBe("No article with that id found.")
+            })
+        })
+
```

Comment:
These are good, you've tested for an invalid id given, and a valid id given which has no results. You've used the correct status codes for these with appropriate response messages, well done.